### PR TITLE
Fix product discount calculation

### PR DESCRIPTION
**Pull request by AliveMen**
### Problem:
The promotion (for special user group)
image

applies to product for ANONYMOUS user 
 
### Solution:
   Fix cart search for anonymous user 
### Changes:
    Fill customerId by new guid while searching anonymous cart

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-6102 has been created

**Review by yecli — APPROVED** [review]

File: src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs [hunk 1]
```diff
@@ -68,10 +69,11 @@ public async Task<CartAggregate> GetCartForShoppingCartAsync(ShoppingCart cart,
 
         public async Task<CartAggregate> GetCartAsync(string cartName, string storeId, string userId, string language, string currencyCode, string type = null, string responseGroup = null)
         {
-            var criteria = new CartModule.Core.Model.Search.ShoppingCartSearchCriteria
+            var criteria = new ShoppingCartSearchCriteria
             {
                 StoreId = storeId,
-                CustomerId = userId,
+                // IMPORTANT! Need to specify customerId, otherwise any user cart could be returned while we expect anonymous in this case.
+                CustomerId = userId ?? AnonymousUserId,
```
Comment (tatarincev) [inline src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs:76]: Could you move this existing type https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/XProfile/VirtoCommerce.XProfile/AnonymousUser.cs into core module and use it constant UserName here?

Comment (tatarincev) [reply src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs:76]: Oh no, Anonymous user has a dependency to Security

Comment (tatarincev) [reply src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs:76]: Better define the constant in core  project then

**Review by yecli — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Pull request merged**